src/Gui/Document.cpp

```diff
@@ -1476,5 +1477,22 @@ void Document::handleChildren3D(ViewProvider* viewProvider)
                 }
             }
         }
+    } else if (viewProvider && viewProvider->isDerivedFrom(ViewProviderDocumentObjectGroup::getClassTypeId())) {

```

Comment:
As we recently introduced extensions this check is not suffiecient. Everything can be a group. You need to check for ViewProvider->hasExtension(ViewProviderDocumentObjectGroup::getExtensionClassTypeId())  This returns true for all Group, GeoFeatureGroup and OriginGroup Viewproviders.